Mail2Bug/Email/EmailBodyProcessingUtils.cs

```diff
@@ -33,21 +33,30 @@ public static string GetLastMessageText_Html(string rawBody)
         {
             CQ dom = rawBody;
 
-            const string messageSeparatorStyle = "border:none;border-top:solid #E1E1E1 1.0pt;padding:3.0pt 0in 0in 0in";
+            const string outlookDesktopSeparatorStyle = "border:none;border-top:solid #E1E1E1 1.0pt;padding:3.0pt 0in 0in 0in";
+            const string outlookMobileSeparatorStyle = "display:inline-block;width:98%";
 
+            // There's no well-defined way to parse the latest email from a thread
+            // We have to use heuristics to cover different email clients
             foreach (IDomObject element in dom["*"])
             {
                 // Lots of email clients insert html elements as message delimiters which have styling but no inner text
                 // This block checks for some of these patterns
-                if (element.NodeName == "DIV")
+                if (string.Equals(element.NodeName, "div", StringComparison.OrdinalIgnoreCase) &&
+                    outlookDesktopSeparatorStyle.Equals(element.GetAttribute("style")))
                 {
-                    if (element.Id == "divRplyFwdMsg" || element.Id == "x_divRplyFwdMsg" || messageSeparatorStyle.Equals(element.GetAttribute("style")))
-                    {
-                        IDomContainer parent = element.ParentNode;
-                        element.Remove();
-                        RemoveSubsequent(parent);
-                        break;
-                    }
+                    IDomContainer parent = element.ParentNode;
+                    RemoveSubsequent(parent);
+                    parent.Remove();
+                    break;
+                }
+
+                if (string.Equals(element.NodeName, "hr", StringComparison.OrdinalIgnoreCase) &&
+                    outlookMobileSeparatorStyle.Equals(element.GetAttribute("style")))
```

Comment:
We should look at a better way to modularize this since we could end up adding more cases. Like a list/map of selectors that we can iterate over. We can also use CSS selectors to pick the right nodes instead of looping through all. Not necessarily to be fixed as part of this PR but something for us to keep a track of.

Reply:
Totally agree, trying to use CSS selectors sounds like a good way to start.